runtime/sema/type.go

```diff
@@ -4491,6 +4499,28 @@ func (t *DictionaryType) initializeMemberResolvers() {
 					)
 				},
 			},
+			"forEachKey": {
+				Kind: common.DeclarationKindFunction,
+				Resolve: func(memoryGauge common.MemoryGauge, identifier string, targetRange ast.Range, report func(error)) *Member {
+					if t.KeyType.IsResourceType() {
```

Comment:
actually wait, do we even allow resources to be keys?

Reply:
No, but this is good future proofing :-)